web/server/entrypoint.go

```diff
@@ -427,3 +436,48 @@ func (s *robotServer) serveWeb(ctx context.Context, cfg *config.Config) (err err
 	}
 	return web.RunWeb(ctx, myRobot, options, s.logger)
 }
+
+// dumpResourceRegistrations prints all builtin resource registrations as a json array
+// to stdout. If you edit this function, ensure that etc/system_manifest/main.go is
+// updated correspondingly.
+func dumpResourceRegistrations() error {
+	type resourceRegistration struct {
+		API             string             `json:"api"`
+		Model           string             `json:"model"`
+		AttributeSchema *jsonschema.Schema `json:"attribute_schema,omitempty"`
+	}
+
+	// create the array of all resource registrations
+	resources := make([]resourceRegistration, 0, len(resource.RegisteredResources()))
+	for apimodel, reg := range resource.RegisteredResources() {
+		var attributeSchema *jsonschema.Schema
+		reflectType := reg.ConfigReflectType()
+		if reflectType != nil {
+			attributeSchema = jsonschema.ReflectFromType(reflectType)
+		}
+		resources = append(resources, resourceRegistration{
+			API:             apimodel.API.String(),
+			Model:           apimodel.Model.String(),
+			AttributeSchema: attributeSchema,
+		})
+	}
+
+	// sort the list alphabetically by API+Model
+	slices.SortFunc(resources, func(a, b resourceRegistration) bool {
+		if a.API != b.API {
+			return a.API < b.API
+		}
+		return a.Model < b.Model
+	})
+
+	// marshall and print the registrations to stdout
+	jsonResult, err := json.Marshal(resources)
+	if err != nil {
+		return errors.Wrap(err, "unable to marshall resources")
+	}
+
+	if _, err := os.Stdout.Write(jsonResult); err != nil {
```

Comment:
As with the other use of Stdout... this feels dangerous. Even more so here. RDK is FULL of things that are eager to write noise to the output. Trying to capture JUST the json cleanly, and making sure no one else EVER makes a change to the code that writes anything else to Stdout when this argument is in use is likely a problem. Would suggest making the "dumpResources" argument take a filename as an option, and (like with the other thing) write a file directly to avoid potential contamination.

Reply:
Technically, there is a test for that case in entrypoint_test.go (which verifies that the output is valid json and matches the registered resources), but I agree that it is a bit strange. Swapping to use a file now.

Reply:
Swapped in 925257c29